src/features/room/sagas.ts

```diff
@@ -0,0 +1,95 @@
+import axios, { AxiosResponse } from 'axios'
+import { Task } from 'redux-saga'
+import {
+  all,
+  call,
+  cancel,
+  delay,
+  fork,
+  put,
+  take,
+  takeLatest,
+} from 'redux-saga/effects'
+
+import {
+  RoomModel,
+  getMatchedRoom,
+  keepRoomAlive,
+} from '../../services/roomService'
+import config from '../../utils/config'
+import { ServiceResponse } from '../../utils/types'
+import { Question } from '../questionBank/types'
+import { closeRoom, openRoom, setQuestionData } from './slice'
+import { RoomSagaActions } from './types'
+
+const keepAliveIntervalInMs = 5000
+const retryFindMatchRoomDelayInMs = 3000
+
+function* startLoadMatchRoom() {
+  const loadRoomTask: Task = yield fork(loadMatchRoomTask)
+  yield take(RoomSagaActions.STOP_LOAD_ROOM)
+  yield cancel(loadRoomTask)
+}
+
+function* loadMatchRoomTask() {
+  while (true) {
+    try {
+      const room: RoomModel = yield getMatchedRoom()
+
+      if (room == null) {
+        throw new Error('Room not found!')
+      }
+
+      yield getQuestionData(room.questionId)
+
+      yield put(openRoom(room))
+    } catch (error) {
+      console.error(`Match room not found! Retrying...`)
```

Comment:
maybe can add a toast here? just so the user is aware that loading the room is still happening or that there was an error

Reply:
Might consider this in the future. For now, the loading is more of a "waiting to connect" state. For example, if the room service fails temporarily or slows down, the component can silently retry the connection without alerting the user. Perhaps can add a toast in the future to provide some feedback along the lines of "Sorry about the delay, Try finding a match".

Reply:
Actually really good suggestion! I added the toast.